### PR TITLE
US114859 - Remove old d2l-offscreen, d2l-link, d2l-loading-spinner, and d2l-icons repos

### PR DESCRIPTION
**Pull request by svanherk**
Remove old `d2l-offscreen`, `d2l-link`, and` d2l-loading-spinner` repos and use the components in `core` instead

Edit: Also removed `d2l-icons` here as well

### CODE REVIEW

File: src/d2l-all-courses.js [hunk 1]
```diff
@@ -288,7 +288,7 @@ class AllCourses extends mixinBehaviors([
 								<d2l-dropdown id="filterDropdown">
 									<button class="d2l-dropdown-opener dropdown-button" aria-labelledby="filterText">
 										<span id="filterText" class="dropdown-opener-text">[[_filterText]]</span>
-										<d2l-icon icon="d2l-tier1:chevron-down" aria-hidden="true"></d2l-icon>
```
Comment (svanherk) [inline src/d2l-all-courses.js:291]: The `d2l-` is no longer needed for `d2l-icon`.  I cleaned these up in `src`, but left them as is in `legacy`

**Comment by svanherk** [comment]
> Is there a reason to keep d2l-link, d2l-icons, and d2l-loading-spinner in the `package.json`?

Hmm I did delete them, are you looking at an old version?

**Comment by AllanKerr** [comment]
> > Is there a reason to keep d2l-link, d2l-icons, and d2l-loading-spinner in the `package.json`?
> 
> Hmm I did delete them, are you looking at an old version?

They are definitely there, not sure how I missed them. 😅

**Comment by svanherk** [comment]
>Is there a reason to keep d2l-link, d2l-icons, and d2l-loading-spinner in the package.json?
>>Hmm I did delete them, are you looking at an old version?
>>> They are definitely removed, not sure how I missed that. 😅

Haha it happens.  After I read your comment I scrolled through the diff and didn't see the file and I was like "Oh wow, did I totally forget to commit the package.json file??" 😝

**Pull request merged**